### PR TITLE
[Debugger] Fix expression evaluation for projects with configurable variables

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR fixes debugger expression evaluations for source projects which includes configurable variables.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/33941

## Approach
Now the debugger runtime uses `io.ballerina.runtime.internal.launch.LaunchUtils#getConfigurationDetails` from the ballerina runtime, to retrieve already resolved configuration details of the original source project and then feeds them into the temporary evaluation project.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/module-info.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
     exports io.ballerina.runtime.internal.values to io.ballerina.testerina.core, io.ballerina.testerina.runtime,
             io.ballerina.lang.xml, org.ballerinalang.debugadapter.runtime;
     exports io.ballerina.runtime.internal.configurable to io.ballerina.lang.internal;
+    exports io.ballerina.runtime.internal.configurable.providers.toml to org.ballerinalang.debugadapter.runtime;
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/module-info.java:66]: It seems, here we need the toml-specific class `TomlDetails` which contains the config file info to call the `$configureInit` method. Do we need to move the class to a more common place? @warunalakshitha

Comment (warunalakshitha) [reply bvm/ballerina-runtime/src/main/java/module-info.java:66]: I think we already expose internal packages to debugger runtime hence this should be ok.

Comment (NipunaRanasinghe) [reply bvm/ballerina-runtime/src/main/java/module-info.java:66]: Yeah all the runtime internals are exposed only to debugger runtime(not to the debug server core).

**Review by HindujaB — APPROVED** [review]
LGTM

**Pull request merged**